packages/@aws-cdk/aws-apigateway/lib/model.ts

```diff
@@ -1,46 +1,141 @@
+   */
+  readonly schema: jsonSchema.JsonSchema;
 }
 
-// TODO: Implement Model, enabling management of custom models.
\ No newline at end of file
+export class Model extends Resource implements IModelRef {
+  /**
+   * Represents a reference to a REST API's Error model, which is available
+   * as part of the model collection by default. This can be used for mapping
+   * error JSON responses from an integration to a client, where a simple
+   * generic message field is sufficient to map and return an error payload.
+   *
+   * Definition
+   * {
+   *   "$schema" : "http://json-schema.org/draft-04/schema#",
+   *   "title" : "Error Schema",
+   *   "type" : "object",
+   *   "properties" : {
+   *     "message" : { "type" : "string" }
+   *   }
+   * }
+   */
+  public static readonly ErrorModel: IModelRef = { modelName: "Error" };
+
+  /**
+   * Represents a reference to a REST API's Empty model, which is available
+   * as part of the model collection by default. This can be used for mapping
+   * JSON responses from an integration to what is returned to a client,
+   * where strong typing is not required. In the absence of any defined
+   * model, the Empty model will be used to return the response payload
+   * unmapped.
+   *
+   * Definition
+   * {
+   *   "$schema" : "http://json-schema.org/draft-04/schema#",
+   *   "title" : "Empty Schema",
+   *   "type" : "object"
+   * }
+   *
+   * @see https://docs.amazonaws.cn/en_us/apigateway/latest/developerguide/models-mappings.html#models-mappings-models
+   */
+  public static readonly EmptyModel: IModelRef = { modelName: "Empty" };
+
+  /** @attribute */
+  public readonly modelName: string;
+
+  /**
+   * The rest API that this model is part of.
+   *
+   * The reason we need the RestApi object itself and not just the ID is because the model
+   * is being tracked by the top-level RestApi object for the purpose of calculating it's
+   * hash to determine the ID of the deployment. This allows us to automatically update
+   * the deployment when the model of the REST API changes.
+   *
+   * @attribute
+   */
+  public readonly restApi: IRestApi;
+
+  constructor(scope: Construct, id: string, props: ModelProps) {
+    super(scope, id, {
+      physicalName: props.modelName || PhysicalName.of(id),
+    });
+
+    if (props.restApi === undefined) {
+      throw Error("You must define a parent rest Api");
+    } else {
+      this.restApi = props.restApi;
+    }
+
+    const modelProps: CfnModelProps = {
+      name: this.physicalName,
+      restApiId: this.restApi.restApiId,
+      contentType: props.contentType,
+      description: props.description,
+      schema: jsonSchema.JsonSchemaMapper.toCfnJsonSchema(props.schema)
+    };
+
+    const resource = new CfnModel(this, 'Resource', modelProps);
+
+    this.modelName = resource.ref;
+
+    const deployment = (this.restApi instanceof RestApi) ? this.restApi.latestDeployment : undefined;
```

Comment:
Why is the `instanceof` check needed here?

Reply:
The `latestDeployment` property is only available in the `RestApi` construct, to ensure I add the model to the hash components, I have to use this property, hence the cast